llama_index/llms/gradient.py

```diff
@@ -0,0 +1,156 @@
+from typing import Any, Optional
+
+from typing_extensions import override
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+from llama_index.callbacks import CallbackManager
+from llama_index.llms import (
+    CompletionResponse,
+    CompletionResponseGen,
+    CustomLLM,
+    LLMMetadata,
+)
+from llama_index.llms.base import llm_completion_callback
+
+
+class _BaseGradientLLM(CustomLLM):
+    _gradient = PrivateAttr()
+    _model = PrivateAttr()
+
+    # Config
+    max_tokens: Optional[int] = Field(
+        description="The number of tokens to generate.",
+        gt=0,
+        lt=512,
```

Comment:
curious: why enforce a limit of 512 output tokens?

Reply:
our current api has the same limit

clipped from our public openapi spec
```
"maxGeneratedTokenCount": {
  "type": "integer",
  "nullable": true,
  "minimum": 0,
  "exclusiveMinimum": true,
  "maximum": 512,
  "exclusiveMaximum": true,
  "description": "The maximum number of tokens to generate."
}
```